<!DOCTYPE html>
<html>
	<head>
		<title>Home page of Abdullah</title>
		<link rel="icon" href="images/abdullah.ico">
		<link rel="stylesheet" href="styles/style.css">
		<script src="https://ajax.googleapis.com/ajax/libs/jquery/3.2.1/jquery.min.js"></script>
		<link rel="stylesheet" href="https://ajax.googleapis.com/ajax/libs/jqueryui/1.12.1/themes/smoothness/jquery-ui.css">
		<script src="https://ajax.googleapis.com/ajax/libs/jqueryui/1.12.1/jquery-ui.min.js"></script>
	</head>

	<body>
		<div id="main-container">
			<!-- Menu -->
			<div id="left-menu">
				<div id="menu-container">
					<a href="#main-container">Intro</a>
					<a href="#work-experience">Work Experience</a>
					<a href="#skills">Programming Skills</a>
					<a href="#projects">Projects</a>
					<a href="#education">Education</a>
					<a href="#achievement">Awards</a>
					<a href="#research-experience">Research Experience</a>
					<a href="#publications">Publications</a>
				</div>
			</div>

			<div id="container">
				<!-- Contact -->
				<div id="header">
					<div id="header-photo">
						<img src="images/abdullah.jpg" id="aam-photo"/>
					</div>
					<div id="header-description">
						<div id="header-name">
							Abdullah-Al Mamun
						</div>
						<div class="separator"></div>
						<div>
							PhD candidate,
							<a href="http://becat.uconn.edu/" target="_blank">Applied algorithms lab</a><br>
							<a href="http://www.cse.uconn.edu/cms/" target="_blank">Computer Science and Engineering Department</a><br>
							<a href="http://www.uconn.edu/" target="_blank">University of Connecticut</a><br>
							<br>
							Contact:<br>
							A38, BECAT, ITEB, University of Connecticut, Storrs, CT-06269.<br>
							<a href="mailto:abdullah.am.cs@engr.uconn.edu">abdullah.am.cs@engr.uconn.edu</a><br>
						</div>
					</div>
				</div>
				<div class="separator"></div>

				<!-- Resource Links -->
				<div id="links">
					<a href="./files/ResumeAbdullahAlMamun.pdf" target="_blank">Resume</a> |
					<a href="https://www.linkedin.com/in/abdullahalmamuncs" target="_blank">Linkedin</a> |
					<a href="https://github.com/abdullah009" target="_blank">GitHub</a>
				</div>
				<div class="separator"></div>

				<!-- Introduction -->
				<div class="description">
					<p>I am a graduate student in Computer Science and Engineering Department at University of
						Connecticut under the supervision of
						<a href="http://www.engr.uconn.edu/~rajasek/" target="_blank">Dr. Sanguthevar Rajasekaran</a>.
						My research area covers applied algorithms, big data analytics, and bioinformatics.
					</p>
				</div>
				<div class="separator"></div>

				<!-- Work Experience -->
				<div class="section" id="work-experience">
					<h2>Work Experience</h2>
					<div class="subsection">
						<div class="double-column">
							<h4>Software Developer Intern</h4>
							<h5>Amazon</h5>
							<h6>Seattle, WA</h6>
						</div>
						<div class="double-column right-column">
							<h5>Jun 2016 - Aug 2016</h5>
						</div>
						<ul>
							<li>
								Designed and implemented authentication services testing application for mobile browsers using JSP, CSS,
								and JavaScript.
							</li>
							<li>
								Extended desktop test application to support A/B testing.
							</li>
						</ul>
					</div>
					<div class="subsection">
						<div class="double-column">
							<h4>Software Engineer</h4>
							<h5>ITIW</h5>
							<h6>Dhaka, Bangladesh</h6>
						</div>
						<div class="double-column right-column">
							<h5>Jan 2011 - Apr 2012</h5>
						</div>
						<ul>
							<li>
								Collaborated closely with other developers to plan, design, and develop popular social network iOS games
								and applications using Objective-C.
							</li>
							<li>
								Built analytics tools in ActionScript3 to monitor performances and tune parameters of products to supply
								better versions.
							</li>
							<li>
								Formulated scalable applications for accessing and modifying billions of user data.
							</li>
							<li>
								Implemented web-based employee tracking software in actionscript3 to record and improve employee activity in workplace.
							</li>
						</ul>
					</div>
					<div class="subsection">
						<div class="double-column">
							<h4>Software Engineer</h4>
							<h5>Aeron Technologies</h5>
							<h6>Dhaka, Bangladesh</h6>
						</div>
						<div class="double-column right-column">
							<h5>May 2009 - Dec 2010</h5>
						</div>
						<ul>
							<li>
								Consulted regularly with clients to provide higher standard software solutions.
							</li>
							<li>
								Wrote maintainable and extensible code in a team environment.
							</li>
							<li>
								Prototyped real-time strategy video games which have been extensively played.
							</li>
							<li>
								Developed more than 20 web-based games in actionscrip3.
							</li>
						</ul>
					</div>
				</div>
				<div class="separator"></div>

				<!-- Programming Skills -->
				<div class="section" id="skills">
					<h2>Skills</h2>
					<div class="subsection">
						<h4>Programming Languages:</h4>
						<label class="name-list">C/C++, Python, Assembly Language, Java, Shell Scripting, ActionScript3</label>
					</div>
					<div class="subsection">
						<h4>Web Development:</h4>
						<label class="name-list">JSP, PHP, JavaScript, CSS, HTML, XML</label>
					</div>
					<div class="subsection">
						<h4>Parallel Programming:</h4>
						<label class="name-list">MPI, OpenMP</label>
					</div>
					<div class="subsection">
						<h4>Database:</h4>
						<label class="name-list">MySQL, Oracle</label>
					</div>
					<div class="subsection">
						<h4>Development Tools:</h4>
						<label class="name-list">UML, E-R Diagram, Matlab, Latex, GNU Plot</label>
					</div>
					<div class="subsection">
						<h4>Operating Systems:</h4>
						<label class="name-list">Unix/Linux, Windows</label>
					</div>
				</div>
				<div class="separator"></div>

				<!-- Projects -->
				<div class="section" id="projects">
					<h2>Projects</h2>
					<div class="subsection">
						<h3><i>k</i>-mer Counting</h3>
						<p>
							A massive number of bioinformatics applications require counting of <i>k</i>-length substrings in
							genetically important long strings. A <i>k</i>-mer counter generates the frequencies of each <i>k
							</i>-length substring in genome sequences. Genome assembly, repeat detection, multiple sequence alignment,
							error detection,and many other related applications use a <i>k</i>-mer counter as a building block. Very
							fast and efficient algorithms are necessary to count <i>k</i>-mers in large data sets to be useful in
							such applications. We propose a novel trie-based algorithm for this <i>k</i>-mer counting problem.
						</p>
						<p>
							We compare our devised algorithm KCMBT (<i>k</i>-mer Counter based on Multiple Burst Trees) with available
							all well-knownw that KCMBT is around 30% faster than the previous bestperforming algorithm KMC2 for
							human genome dataset. As another example, our algorithm is around six times faster than Jellyfish2.
							Overall, KCMBT is 20%-30% faster than KMC2 on five benchmark data sets when both the algorithms were
							run using multiple threads.
						</p>
						<p>
							<a href="https://github.com/abdullah009/kcmbt_mt" target="_blank">KCMBT</a> is freely available on
							GitHub.
						</p>
					</div>
					<div class="subsection">
						<h3>Record Linkage</h3>
						<p>
							Integrating data from multiple sources is a crucial and challenging problem. Even though there exist
							numerous algorithms for record linkage or deduplication, they suffer from either large time needs or
							restrictions on the number of datasets that they can integrate. We report efficient sequential and
							parallel algorithms for record linkage which handle any number of datasets and outperform previous
							algorithms. Our algorithms employ hierarchical clustering algorithms as the basis. A key idea
							that we use is radix sorting on certain attributes to eliminate identical records before any further
							processing. Another novel idea is to form a graph that links similar records and find the connected
							components.
						</p>
						<p>
							Our sequential and parallel algorithms have been tested on a real dataset of 1,083,878 records and
							synthetic datasets ranging in size from 50,000 to 9,000,000 records. Our sequential algorithm runs at
							least two times faster, for any dataset, than the previous best-known algorithm, the two-phase
							algorithm using faster computation of the edit distance (TPA (FCED)). The speedups obtained by our
							parallel algorithm are almost linear. For example, we get a speedup of 7.5 with 8 cores (residing in a
							single node), 14.1 with 16 cores (residing in two nodes), and 26.4 with 32 cores (residing in four nodes).
						</p>
						<p>
							The above algorithms use single linkage-based hierarchical clustering. This linkage comes with chaining
							problem. To resolve this issue, we propose efficient as well	as reliable sequential and parallel
							complete linkage hierarchical clustering algorithms. We have tested our algorithms on datasets with
							millions of synthetic records. Experimental results show that our algorithms achieve nearly 100%
							accuracy. Parallel implementations achieve almost linear speedups.
						</p>
						<p>
							Huge achievement in record linkage methods has pushed us to make them freely available for interested
							agencies. RLT-S tool (<a href="http://www.rlatools.com" target="_blank">www.rlatools
							.com</a>) implements our record linkage algorithm. This website also contains necessary
							information such as instructions, submission history, feedback, publications and some other sections to
							facilitate the usage of the tool. The <a href="https://github.com/abdullah009/RLATools"
							                                         target="_blank">source code</a> is publicly available on GitHub.
						</p>
					</div>
					<div class="subsection">
						<h3>Higher-order Spectra Analysis</h3>
						<p>
							Polyspectral estimation is a problem of great importance in the analysis of nonlinear time series that
							has applications in communications, sonar, radar, speech, and image processing, biomedical signal
							processing, geophysics, etc. For instance, higher order spectra have been used in unsupervised and
							supervised clustering in big data scenarios, as a test for Gaussianity, to suppress Gaussian noise, to
							characterize nonlinearities in any data, and so on. Any algorithm for computing the <i>k</i>th order
							spectra on a dataset of size <i>n</i> needs <i>&#937(nk - 1)</i> time since the output size will be
							<i>&#937(nk - 1)</i> as well. Given that we live in an era of big data, <i>n</i> could be very large. In
							this case, sequential algorithms might take unacceptable amounts of time. Thus it is essential to
							develop parallel algorithms. We offer generic sequential algorithms for computing higher order spectra
							(HOS) that are asymptotically faster than any published algorithm for HOS. In addition, we offer memory
							efficient algorithms. Also, we present optimal parallel implementations of these algorithms on parallel
							computing models such as the PRAM and the mesh. We also provide experimental results on our sequential
							and parallel algorithms. Our parallel implementation on a CPU cluster achieves good speedups.
						</p>
					</div>
					<div class="subsection">
						<h3>Minimum Spanning Tree Problem</h3>
						<p>
							Finding minimum spanning trees (MST) in various types of networks is a well-studied problem in theory
							and practical applications. A number of efficient algorithms have been already developed for this
							problem. We present an efficient algorithm, namely Edge Pruned Minimum Spanning Tree (EPMST) algorithm,
							which combines ideas from randomized selection, Kruskal's algorithm and Prim's algorithm. The algorithm
							has a superior performance relative to the well-known algorithms especially when the graph is not very
							sparse.
						</p>
					</div>
					<div class="subsection">
						<h3>Closest <i>l</i>-mers</h3>
						<p>
							With the advances in the next generation sequencing technology, huge amounts of data have been
							generated in biology. A bottleneck in dealing with such datasets lies in developing effective
							algorithms for extracting useful information from them. Algorithms for finding patterns in biological
							data pave the way for extracting crucial information from voluminous datasets. In this project we focus
							on a fundamental pattern, namely, the closest <i>l</i>-mers. Given a set of <i>m</i> biological strings
							<i>S<sub>1</sub></i>, <i>S<sub>2</sub></i>, ..., <i>S<sub>m</sub></i> and an integer <i>l</i>, the
							problem of interest is that of finding an <i>l</i>-mer from each string such that the distance among
							them is the least. This problem has many applications. An application of great importance is motif
							search. Algorithms for finding the closest <i>l</i>-mers have been used in solving the <i>(l, d)</i>-
							motif search problem. Here we have developed exact and approximate algorithms for this problem for the
							special case of <i>m = 3</i>. We consider the Euclidean distance metric if the sequences contain real
							numbers.
						</p>
					</div>
				</div>
				<div class="separator"></div>

				<!-- Education -->
				<div class="section" id="education">
					<h2>Education</h2>
					<div class="subsection">
						<div class="double-column">
							<h4>Doctor of Philosophy in Computer Science and Engineering</h4>
							<h5>University of Connecticut</h5>
							<h6>Storrs, CT</h6>
						</div>
						<div class="double-column right-column">
							<h5>May 2018</h5>
							<h6>3.91 out of 4.0</h6>
						</div>
					</div>
					<div class="subsection">
						<div class="double-column">
							<h4>Bachelor of Science in Computer Science and Engineering</h4>
							<h5>Bangladesh University of Engineering and Technology</h5>
							<h6>Dhaka, Bangladesh</h6>
						</div>
						<div class="double-column right-column">
							<h5>Mar 2009</h5>
							<h6>3.85 out of 4.0</h6>
						</div>
					</div>
				</div>
				<div class="separator"></div>

				<!-- Awards -->
				<div class="section" id="achievement">
					<h2>Awards</h2>
					<ul>
						<li>
							Predoctoral Fellowship Award (2013, 2014, 2015, 2016, 2017), UConn.
						</li>
						<li>
							Dean's List Awards in BUET.
						</li>
						<li>
							University Merit Scholarships in BUET.
						</li>
						<li>
							18th place in entrance exam of BUET.
						</li>
					</ul>
				</div>
				<div class="separator"></div>

				<!-- Research Experience -->
				<div class="section" id="research-experience">
					<h2>Research Experience</h2>
					<div class="subsection">
						<div class="double-column">
							<h4>Graduate Research Assistant</h4>
							<h5>University of Connecticut</h5>
							<h6>Storrs, CT</h6>
						</div>
						<div class="double-column right-column">
							<h5>Aug 2012 - Present</h5>
						</div>
						<ul>
							<li>
								Devise algorithms for big data, high performance computing, distributed systems.
							</li>
							<li>
								Improve algorithm performances via coding effectively, playing with advanced data structures, and
								evaluating interface between hardware and software.
							</li>
						</ul>
					</div>
				</div>
				<div class="separator"></div>

				<!-- Publications -->
				<div class="section" id="publications">
					<h2>Publications</h2>
					<ol class="ordered-list">
						<li class="list-item">
							Xingyu Cai&#42;, <b>Abdullah-Al Mamun</b>&#42;, and Sanguthevar Rajasekaran. Efficient Algorithms for
							Finding the Closest <i>l</i>-mers in Biological Data. IEEE/ACM Transactions on Computational Biology and
							Bioinformatics (TCBB), 2017 (submitted).
						</li>
						<li class="list-item">
							<b>Abdullah-Al Mamun</b>, Xingyu Cai, Zigeng Wang, Nalini Ravishanker, and Sanguthevar Rajasekaran. Efficient
							Sequential and Parallel Algorithms for Estimating Higher Order Spectra. IPDPS, 2018 (submitted).
						</li>
						<li class="list-item">
							Kenneth F. Lyon, Xingyu Cai, Richard J. Young, <b>Abdullah-Al Mamun</b>, Sanguthevar Rajasekaran, and
							Martin R. Schiller. Minimotif Miner 4: a Million Peptide Minimotifs and Counting. Nucleic Acids Research,
							2017 (accepted).
						</li>
						<li class="list-item">
							Xingyu Cai&#42;, <b>Abdullah-Al Mamun</b>&#42;, and Sanguthevar Rajasekaran. Novel Algorithms for Finding the
							Closest <i>l</i>-mers in Biological Data. BIBM, 2017 (accepted).
						</li>
						<li class="list-item">
							<b>Abdullah-Al Mamun</b>, and Sanguthevar Rajasekaran.
							<a href="http://ieeexplore.ieee.org/abstract/document/7543874/" target="_blank">An efficient Minimum
								Spanning Tree algorithm</a>. IEEE Symposium on Computers and Communication (ISCC), 2016.
						</li>
						<li class="list-item">
							<b>Abdullah-Al Mamun</b>, Soumitra Pal, and Sanguthevar Rajasekaran.
							<a href="https://academic.oup.com/bioinformatics/article/32/18/2783/1744340" target="_blank">
								KCMBT: a <i>k</i>-mer Counter based on Multiple Burst Trees
							</a>. Bioinformatics, 2016.
						</li>
						<li class="list-item">
							<b>Abdullah-Al Mamun</b>, Robert Aseltine, and Sanguthevar Rajasekaran.
							<a href="http://journals.plos.org/plosone/article?id=10.1371/journal.pone.0154446" target="_blank">
								Efficient Record Linkage Algorithms Using Complete Linkage Clustering
							</a>. PLOS ONE, 2016.
						</li>
						<li class="list-item">
							<b>Abdullah-Al Mamun</b>, Robert Aseltine, and Sanguthevar Rajasekaran.
							<a href="http://journals.plos.org/plosone/article?id=10.1371/journal.pone.0124449" target="_blank">
								RLT-S: A Web System for Record Linkage
							</a>.
							PLOS ONE, 2015.
						</li>
						<li class="list-item">
							<b>Abdullah-Al Mamun</b>, Tian Mi, Robert Aseltine, Sanguthevar Rajasekaran,
							<a href="https://academic.oup.com/jamia/article/21/2/252/2909251" target="_blank">
								Efficient sequential and parallel algorithms for record linkage
							</a>.
							Journal of the American Medical Informatics Association (JAMIA), 2014.
						</li>
					</ol>
					&#42; Co-first authors with equal contributions to this work
				</div>

				<div class="separator"></div>
				<div id="footer">
				</div>
			</div>

			<div id="right-menu">

			</div>
		</div>
	</body>
</html>
